xonsh/lazyimps.py

```diff
@@ -88,3 +89,15 @@ def terminal256():
 @lazyobject
 def html():
     return importlib.import_module("pygments.formatters.html")
+
+
+@lazyobject
+def os_listxattr():
+    if getattr(os, "listxattr", None):
```

Comment:
I'm not a fan of `hasattr()` [e.g](https://hynek.me/articles/hasattr/), although I'd agree `os.listxattr` probably doesn't have any of those issues.

I'll go with the second alternative, though it does define a redundant closure.  It's good!.

Reply:
Thanks for the reading. So it seems that the problem is mostly py2, which is AFAIK not supported anymore by xonsh?

If I understand the problem correctly, it is that py2's hasattr accesses the attribute instead of just looking it up. Then your original code does not solve the issue, because you threw away the result of getattr... Something like would already be better:

```
xx = getattr(os, "listxattr", None)
if xx is None:
    def xx...
return xx